### PR TITLE
Harmonized URLs and Methods

### PR DESCRIPTION
**Pull request by iannjari**
Each URL has it's GET, POST, PUT, DELETE. 

The URLs are:
`/users`,

`/tasks`

and 

`/users/auth`   ---> since `users` can't hold more than one `@GetMapping`

Review and suggest changes.

### CODE REVIEW

File: endpoints/endpoints.md [hunk 1]
```diff
@@ -12,75 +14,81 @@ mock_responses:
...
-8. HTTP: `/update_user` : method: `PUT` parameters: `(id,username,password,email,display_name)` response: `{[updated:true/false]}` description: updates user details
+- `PUT` method: parameters: `(id,title,description,due_date)` response: `{[updated:true/false]}` description: updates task details
 
 mock_response:
 ```json
 {"updated":true,}
 ```
 
-9. HTTP: `/delete_task` : method: `DELETE` parameters: `(id)` response: `{[deleted:true/false]}` description: deletes task
+- `DELETE` method: parameters: `(id)` response: `{[deleted:true/false]}` description: deletes task
```
Comment (JaelNgutu) [inline endpoints/endpoints.md]: @iannjari  we can return a string "task deleted successfully"

File: endpoints/endpoints.md [hunk 2]
```diff
@@ -12,75 +14,81 @@ mock_responses:
...
-mock_response:
-```json
-{"username":"ernnjf",
- "email":"x.y@gmail.com",
- "display_name":"Abc Def",
- "profile_photo":"https://dummyimage.com/161x100.png"}
- ```
- 
-8. HTTP: `/update_user` : method: `PUT` parameters: `(id,username,password,email,display_name)` response: `{[updated:true/false]}` description: updates user details
+- `PUT` method: parameters: `(id,title,description,due_date)` response: `{[updated:true/false]}` description: updates task details
```
Comment (JaelNgutu) [inline endpoints/endpoints.md]: @iannjari  he we should return the updated task  ie {id,title,description,due_date}

File: endpoints/endpoints.md (hunk 1 above)
Comment (iannjari) [reply endpoints/endpoints.md]: Okay, lemme implement this then merge

File: endpoints/endpoints.md (hunk 2 above)
Comment (iannjari) [reply endpoints/endpoints.md]: Sure!

**Comment by iannjari** [comment]
[Update API endpoints.md - Backend](https://trello.com/c/sXbAVDNy/32-update-api-endpointsmd-backend)

**Pull request merged**